griptape/utils/structure_visualizer.py

```diff
@@ -34,7 +38,14 @@ def to_url(self) -> str:
         graph_bytes = graph.encode("utf-8")
         base64_string = base64.b64encode(graph_bytes).decode("utf-8")
 
-        return f"https://mermaid.ink/svg/{base64_string}"
+        url = urllib.parse.urljoin(self.base_url, f"svg/{base64_string}")
+        req = PreparedRequest()
+        req.prepare_url(url, self.query_params)
+
+        if req.url is None:
+            raise ValueError("Failed to generate the URL")
```

Comment:
I can't for the life of me figure out how to trigger this, but `pyright` says it's `str | None` so 🤷